test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -222,32 +221,13 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
 				_, err := cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
 				framework.ExpectNoError(err)
 
-				// Get new copy of the snapshot
-				ginkgo.By("checking the snapshot")
-				vs, err = dc.Resource(storageutils.SnapshotGVR).Namespace(vs.GetNamespace()).Get(context.TODO(), vs.GetName(), metav1.GetOptions{})
-				framework.ExpectNoError(err)
-
-				// Get the bound snapshotContent
-				snapshotStatus := vs.Object["status"].(map[string]interface{})
-				snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
-				vscontent, err := dc.Resource(storageutils.SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
-				framework.ExpectNoError(err)
-
-				snapshotContentSpec := vscontent.Object["spec"].(map[string]interface{})
-				volumeSnapshotRef := snapshotContentSpec["volumeSnapshotRef"].(map[string]interface{})
+				vs := sr.Vs
```

Comment:
in the previous code `vs` is updated for the rest of the function with the latest version from the API server

```
vs, err = dc.Resource(storageutils.SnapshotGVR).Namespace(vs.GetNamespace()).Get(context.TODO(), vs.GetName(), metav1.GetOptions{})
```

in the refactor that update only happens within the helper function, luckily, the properties of `vs` in this function that came from the update aren't used so it should work